bitnami/redis/templates/health-configmap.yaml

```
@@ -12,35 +12,49 @@ metadata:
   {{- end }}
 data:
   ping_readiness_local.sh: |-
-    #!/bin/bash
+    #!/usr/bin/env bash
+    REDIS_HOST="localhost"
+    REDIS_PORT="6379"
+    redis_cli=(timeout -s 3 1 redis-cli -h "$REDIS_HOST" -p "$REDIS_PORT")
 
     [[ -f $REDIS_PASSWORD_FILE ]] && export REDIS_PASSWORD="$(< "${REDIS_PASSWORD_FILE}")"
     [[ -n "$REDIS_PASSWORD" ]] && export REDISCLI_AUTH="$REDIS_PASSWORD"
-    response=$(
-      timeout -s 3 $1 \
-      redis-cli \
-        -h localhost \
-{{- if .Values.tls.enabled }}
-        -p $REDIS_TLS_PORT \
-        --tls \
-        --cacert {{ template "redis.tlsCACert" . }} \
-        {{- if .Values.tls.authClients }}
-          --cert {{ template "redis.tlsCert" . }} \
-          --key {{ template "redis.tlsCertKey" . }} \
-        {{- end }}
-{{- else }}
-        -p $REDIS_PORT \
-{{- end }}
-        ping
-    )
-    if [ "$?" -eq "124" ]; then
-      echo "Timed out"
-      exit 1
-    fi
-    if [ "$response" != "PONG" ]; then
-      echo "$response"
-      exit 1
-    fi
+
+    function getRedisRole() {
+        "${redis_cli[@]}" info replication 2>/dev/null | grep "role" | tr -d "\r" | tr -d "\n"
```

Isn't `redis info replication` specific to `--set architecture=replication` option for the chart (and not `standalone`)?

In addition, a small improvement that could be made:

```suggestion
        "${redis_cli[@]}" info replication 2>/dev/null | grep -Eo "^role:.*"
```

(please also apply this below)